src/cwe_checker_lib/src/intermediate_representation/project/propagate_control_flow.rs

```diff
@@ -242,15 +256,44 @@ fn negate_condition(expr: Expression) -> Expression {
     }
 }
 
+/// Iterates the CFG and returns all node's blocks, that do not have an incoming edge.
+fn get_nodes_without_incomming_edge(cfg: &Graph) -> HashSet<Tid> {
+    let mut nodes_without_incomming_edges = HashSet::new();
+    for node in cfg.node_indices() {
+        if cfg.neighbors_directed(node, Incoming).next().is_none() {
+            println!("{}", cfg[node].get_block().tid.clone());
+            nodes_without_incomming_edges.insert(cfg[node].get_block().tid.clone());
+        }
+    }
+    nodes_without_incomming_edges
+}
+
+/// Calculates the difference of the orphaned blocks and removes them from the project.
+fn remove_new_orphaned_blocks(
+    project: &mut Project,
+    orphaned_blocks_before: HashSet<Tid>,
+    orphaned_blocks_after: HashSet<Tid>,
+) {
+    let new_orphan_blocks = orphaned_blocks_after
+        .difference(&orphaned_blocks_before)
+        .collect_vec();
```

Comment:
Since you call `new_orphan_blocks.contains(..)` a lot, it is probably faster to collect these blocks in a `HashSet<Tid>` instead of a vector.

Reply:
Very good point! Thank you.